[ACA-4679] Added code changes and env variables to enable DownloadPrompt and FileAutoDownload features on ACA

swapnil-verma-gl opened the pull request:
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:


**What is the current behaviour?** (You can also link to an open issue here)
The DownloadPrompt and FileAutoDownload features implemented and ADF are not completely visible on the ACA application. This is due to data table column overrides and missing env variable configurations


**What is the new behaviour?**
Updated the data table column overrides and added env variable configurations to enable the DownloadPrompt and the FileAutoDownload feature on ACA


**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No


If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
More information on this feature can be found in the following JIRAs - 
- [ACA-4675 (Epic)](https://alfresco.atlassian.net/browse/ACA-4675)
- [ACA-4679 (Story)](https://alfresco.atlassian.net/browse/ACA-4679)

MichalKinas reviewed (approved):
Changes LGTM, but please rebase your PR to get newest license changes and add licenses to newly added files as linter suggest

swapnil-verma-gl commented:
> Changes LGTM, but please rebase your PR to get newest license changes and add licenses to newly added files as linter suggest

Thank you @MichalKinas. Rebased the PR and added the missing license headers.

DenysVuika commented on .github/actions/run-e2e/action.yml:
```diff
@@ -37,6 +37,12 @@ runs:
           echo "APP_CONFIG_OAUTH2_REDIRECT_LOGOUT=/"
           echo "APP_CONFIG_OAUTH2_REDIRECT_LOGIN=/"
           echo "APP_CONFIG_OAUTH2_REDIRECT_SILENT_IFRAME_URI={protocol}//{hostname}{:port}/assets/silent-refresh.html"
+          echo "APP_CONFIG_ENABLE_DOWNLOAD_PROMPT=true"
```
update main readme file with these variables

swapnil-verma-gl replied:
Updated the README.md file. The values I added to that file are -

APP_CONFIG_ENABLE_DOWNLOAD_PROMPT=true
APP_CONFIG_ENABLE_DOWNLOAD_PROMPT_REMINDERS=true
APP_CONFIG_DOWNLOAD_PROMPT_DELAY=`<time>`
APP_CONFIG_DOWNLOAD_PROMPT_REMINDER_DELAY=`<time>`
APP_CONFIG_ENABLE_FILE_AUTO_DOWNLOAD=true
APP_CONFIG_FILE_AUTO_DOWNLOAD_SIZE_THRESHOLD_IN_MB=`<file-size>`

Please validate if these are correct or if we should add the actual values instead.

DenysVuika commented:
Instead of the angular service that you now have to provide in each constructor, you can create the Effect like in all other cases, and use store.dispatch

DenysVuika commented on projects/aca-content/src/lib/components/toolbar/view-node/view-node.component.ts:
```diff
@@ -53,22 +54,26 @@ import { SharedLinkEntry } from '@alfresco/js-api';
-        if (selection.file.entry.nodeType === 'app:filelink') {
-          id = selection.file.entry.properties['cm:destination'];
+        if (this.fileAutoDownloadService.shouldFileAutoDownload(selection.file.entry?.content?.sizeInBytes)) {
+          this.fileAutoDownloadService.autoDownloadFile(selection.file);
```
there's already store effects in this file, so better having an effect that a separate service

swapnil-verma-gl replied:
The service currently performs 2 functions - Evaluate whether a file needs to downloaded or not (done via `service.shouldFileAutoDownload(fileSizeInBytes)` method), and open the prompt to download the file in case the first method returns true (done via `service.autoDownloadFile(node)` method). In case the first method returns a false value, the application should behave normally as it has been so far and execute the code that was present before. 

I understand that it is easy enough to move the second functionality inside a store effect, but how can we move the first one while making sure we have a back reference to execute the existing code in case the file should not be downloaded?

DenysVuika commented on projects/aca-shared/src/lib/services/aca-file-auto-download.service.ts:
```diff
@@ -0,0 +1,51 @@
+@Injectable({
+  providedIn: 'root'
+})
+export class AcaFileAutoDownloadService {
```
there's already a https://github.com/Alfresco/alfresco-content-app/blob/4956655a801bdb714fa76399df0fe81d8c5976c0/projects/aca-content/src/lib/store/effects/download.effects.ts#L37 containing various effects for dowloading
